httpx/models.py

```diff
@@ -688,6 +688,20 @@ def __init__(
             self.is_stream_consumed = False
             self._raw_stream = stream
 
+    @property
+    def elapsed(self) -> datetime.timedelta:
+        """
+        Returns the time taken for the complete request/response
+        cycle to complete.
+        """
+        if not hasattr(self, "_elapsed"):
+            msg = (
+                "`elapsed` may only be called after calling `close()`"
+                " on the response"
+            )
+            raise AssertionError(msg)
```

Comment:
Perhaps this would be better served by a `RuntimeError`? Also some stylistic and copy suggestions:

```python
raise RuntimeError(
    "'.elapsed' may only be accessed after the response "
    "has been read or closed."
)
```

Reply:
Fixed.